### DISCUSSION TITLE
How can I sort customer transactions on transaction date?

### DISCUSSION THREAD

**Discussion by iKeysa** [dynamoose/dynamoose#974] [Q&A] [answered]
Below is my table structure

- customer (hashkey)
- transactionType & transactionId (range key)
- transactionDate
- completionStatus

Need to support the following to display data on webpage.

Return all customer transactions in a specific time period order by transaction date (either ascending or descending). Additionally consumer can request to return transactions which are completed/in complete/both. Trying to build the query as follows, from documentation sort order is done on range key, is there any way to sort based on transactionDate?

const sortOrder = "ascending";
const limit = 30;
const query: any =Model.query(customer).eq(id)
        .where(rangeKey).beginsWith ("transaction1").and()
        .where(completionStatus).eq (false).and()
        .where(transactionDate).between(1595448622049,1595448622049); // in milliseconds
await query.limit(limit ).sort(sortOrder).exec();

**Comment by fishcharlie** [comment, marked as answer]
@iKeysa Sadly I haven't developed a good system handling discussions yet. I will make an announcement when the feature is ready for us for Dynamoose. In the meantime, please use the Slack channel or create an issue on the repo (as opposed to a discussion).

In regards to your question. No. This is a limit with DynamoDB. As mentioned in the [documentation](https://dynamoosejs.com/guide/Query#querysortorder):

> This function sorts the documents you receive back by the rangeKey.

So in order to achieve this you likely have to rethink and redesign your table structure and how you are doing this.

**Comment by pseudoCoder0** [comment]
@iKeysa if you have used UUID (recommended) for your range key then it is time based id, which is in sorted order. We came across similar situation and found the solution by sorting the range key as UUID is being used.

Hope this helps! Thanks!
